packages/ui/src/Tooltip/UseTooltip.tsx

```diff
@@ -8,36 +8,71 @@ type Position = {
 type UseTooltipProperties = {
   ref: RefObject<HTMLElement>;
   tooltipReference: RefObject<HTMLDivElement>;
+  position?: "top" | "bottom" | "right" | "left";
 };
 
-export function useTooltip({ ref, tooltipReference }: UseTooltipProperties) {
+export function useTooltip({
+  ref,
+  tooltipReference,
+  position,
+}: UseTooltipProperties) {
   const [showTooltip, setShowTooltip] = useState<boolean>(false);
-  const [position, setPosition] = useState<Position>({});
+  const [tooltipPosition, setTooltipPosition] = useState<Position>({});
 
   useEffect(() => {
     if (!ref.current) {
       return;
     }
 
     if (showTooltip) {
-      const { left, width, top } = ref.current.getBoundingClientRect();
+      const { left, right, top, bottom, height, width } =
+        ref.current.getBoundingClientRect();
 
       const tooltipWidth =
         tooltipReference?.current?.getBoundingClientRect().width || 0;
 
       const tooltipHeight =
         tooltipReference?.current?.getBoundingClientRect().height || 0;
 
-      const middle = left + width / 2 - tooltipWidth / 2;
+      const horizontalCenter = left + width / 2 - tooltipWidth / 2;
 
-      setPosition({
-        top: top - tooltipHeight,
-        left: middle,
-      });
+      const verticalCenter = top + height / 2 - tooltipHeight / 2;
+
+      switch (position) {
+        case "top":
+          setTooltipPosition({
+            top: top - tooltipHeight - 5,
```

Comment:
what is 5 here and in below calculations? Should we define a constant for it? Or maybe make it configurable? Or if possible have it defined for css which allows for easy customization?